contracts/strategy/LiquidationStrategy.sol

```diff
@@ -2,83 +2,118 @@
+import { ILiquidationStrategy } from "./interfaces/liquidationStrategy/ILiquidationStrategy.sol";
+import { IERC165 } from "./interfaces/liquidationStrategy/IERC165.sol";
+import { IGenericTokenAdapter } from "./interfaces/liquidationStrategy/IGenericTokenAdapter.sol";
+import { IUniswapV2Factory } from "./interfaces/liquidationStrategy/IUniswapV2Factory.sol";
+import { IUniswapV2Router02 } from "./interfaces/liquidationStrategy/IUniswapV2Router02.sol";
+import { IStablecoinAdapter } from "./interfaces/liquidationStrategy/IStablecoinAdapter.sol";
 import { IBookKeeper } from "./interfaces/liquidationStrategy/IBookKeeper.sol";
-import "./libraries/BytesHelper.sol";
+import { BytesHelper } from "./libraries/BytesHelper.sol";
 
 /// @title LiquidationStrategy for FathomVault
 /// @notice Enables participation in generating profits from liquidations and contributes to the liquidation of FXD positions.
 /// @dev Inherits from BaseStrategy, ReentrancyGuard, implements IFlashLendingCallee, and IERC165.
 
 // solhint-disable
-contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCallee, IERC165 {
+contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCallee, ILiquidationStrategy, IERC165 {
     using SafeERC20 for ERC20;
     using SafeMath for uint256;
     using BytesHelper for *;
 
     struct LocalVars {
         address liquidatorAddress;
         IGenericTokenAdapter tokenAdapter;
-        IUniswapV2Router02 router;
+        address routerV2;
+        address routerV3;
+        uint256 v2Ratio; // bps 10000 = 1%
     }
 
-    struct WXDCInfo {
-        uint256 WXDCAmount;
+    struct CollateralInfo {
+        uint256 CollateralAmount;
         uint256 amountNeededToPayDebt;
-        uint256 averagePriceOfWXDC;
+        uint256 averagePriceOfCollateral;
     }
 
+    struct UniswapV3Info {
+        address permit2;
+        address universalRouter;
+        uint24 poolFee;
+    }
+
+    // Command for V3_SWAP_EXACT_IN
+    bytes1 constant COMMAND = 0x00; // Assuming 0x00 is the COMMAND for V3_SWAP_EXACT_IN
+
     // --- Math ---
     uint256 constant WAD = 10 ** 18;
     uint256 constant RAY = 10 ** 27;
+    uint256 constant TENK = 10000;
 
     IStablecoinAdapter public stablecoinAdapter;
     IBookKeeper public bookKeeper;
     address public strategyManager;
     address public fixedSpreadLiquidationStrategy;
-    ERC20 public WXDC;
     ERC20 public fathomStablecoin;
     ERC20 public usdToken;
-    bool public allowLoss = true;
-    WXDCInfo public idleWXDC;
-
-    event LogSetStrategyManager(address indexed _strategyManager);
-    event LogSetFixedSpreadLiquidationStrategy(address indexed _fixedSpreadLiquidationStrategy);
-    event LogShutdownWithdrawWXDC(address indexed _strategyManager, uint256 _amount);
-    event LogAllowLoss(bool _allowLoss);
-    event LogSellWXDC(
+
+    mapping(address => CollateralInfo) public idleCollateral;
+    mapping(address => UniswapV3Info) public uniswapV3Info;
```

Comment:
here, we store the universalRouter address twice: as a key and inside UniswapV3Info, which is redundant